caddy/caddymain/run.go

```diff
@@ -409,13 +415,70 @@ func initTelemetry() error {
 	return nil
 }
 
+// LoadEnvFromFile loads additional envs if file provided and exists
+//Envs in file should be in KEY=VALUE format
+func LoadEnvFromFile(envFile string) error {
+	if envFile == "" {
+		return nil
+	}
+
+	file, err := os.Open(envFile)
+
+	if err != nil {
+		return err
+	}
+	defer file.Close()
+
+	envMap := make(map[string]string)
+
+	scanner := bufio.NewScanner(file)
+	var line string
+
+	for scanner.Scan() {
+		line = scanner.Text()
+
+		//skip lines starting with comment
+		if strings.HasPrefix(line, "#") {
+			continue
+		}
+
+		//skip empty line
+		if len(line) == 0 {
+			continue
+		}
+
+		fields := strings.SplitN(line, "=", 2)
+		if len(fields) != 2 {
+			return errors.New("The line should be in KEY=VALUE format")
```

Comment:
For this and other occurrences, use `fmt.Errorf` and give the line number of the error.